Add wording to integrate discussions from p2581r2 (#22)
cplusplus/modules-ecosystem-tr: buildprocess.tex

\indextext{build|(}%

\rSec1[build.import]{Import relationship between translation units}

\pnum The name of a module used in the ``import'' and ``module''
keywords is the ``logical name'' of a module.

\pnum When a translation unit imports a named module, the translation
unit doing the import has an import dependency on the translation unit
providing the importable unit identified via the ``logical name''.

\pnum The translation of an importable unit is externalized into an
intermediate artifact, the Built Module Interface file (BMI). That
file is then consumed when importing a named module.

> \rSec1[build.arguments]{Compiler arguments and the import relationship}
> 
> \pnum ``ABI coherency'' is the expected level of coherency in the
> compilation command for the translation of the code in the purview of a
> module and all translation units importing that module. When that
> level of coherency is not met, the final program may fail to link or
> it may link and have undefined behavior at runtime.
> 
> \pnum ``BMI coherency'' is the expected level of coherency in the
> compiler arguments for a BMI file to be used when importing a

Reviewer: ```suggestion
\rSec1[build.arguments]{Compilation commands and the import relationship}

\pnum ``ABI coherency'' is the expected level of coherency in the
compilation command for the translation of the code in the purview of a
module and all translation units importing that module. When that
level of coherency is not met, the final program may fail to link or
it may link and have undefined behavior at runtime.

\pnum ``BMI coherency'' is the expected level of coherency in the
compilation command for a BMI file to be used when importing a
```

\indextext{build|(}%

\rSec1[build.import]{Import relationship between translation units}

\pnum The name of a module used in the ``import'' and ``module''
keywords is the ``logical name'' of a module.

\pnum When a translation unit imports a named module, the translation
unit doing the import has an import dependency on the translation unit
providing the importable unit identified via the ``logical name''.

\pnum The translation of an importable unit is externalized into an
intermediate artifact, the Built Module Interface file (BMI). That
file is then consumed when importing a named module.

> 
> \pnum BMI files can be constructed in a way to optimize for the
> efficiency of the reuse in the import operation, which may result in a
> limitation on the number of scenarios where that file can be used.

Reviewer: ```suggestion
```

> \pnum BMI files can be constructed in a way to optimize for the
> efficiency of the reuse in the import operation, which may result in a
> limitation on the number of scenarios where that file can be used.

Reviewer 2: ```suggestion
\pnum A BMI file can be constructed to optimize for
efficient reuse in an import operation, which may result in a
limitation on the number of scenarios where that file can be used.
```

\rSec1[build.arguments]{Compiler arguments and the import relationship}

\pnum ``ABI coherency'' is the expected level of coherency in the

> compiler arguments for the translation of the code in the purview of a

Reviewer: ```suggestion
compilation command for the translation of the code in the purview of a
```

module and all translation units importing that module. When that
level of coherency is not met, the final program may fail to link or
it may link and have undefined behavior at runtime.

\pnum ``BMI coherency'' is the expected level of coherency in the
compiler arguments for a BMI file to be used when importing a
module. When that level of coherency is not met, the translation will
fail because the BMI cannot be loaded.

> 

Reviewer 2: Suggestion: Examples where ABI and BMI coherency are and are not met would be helpful.

Reviewer: Created https://github.com/cplusplus/modules-ecosystem-tr/issues/25 to follow up on this

\pnum While all translation units importing a module must have ``ABI
coherency'', different translation units importing the same module may
not meet the requirements for ``BMI coherency''. It must be valid for
those to consume different BMI files of the same importable unit, and
to be linked into the same program.

\pnum ``Local Preprocessor Arguments'' are the arguments that are
expected to be different in the translation of an importable unit and
the translation of a unit importing that named module.

\pnum The decision on whether an argument is a ``Local Preprocessor
Argument'' cannot be made by introspection on the code or the command

> line, and should be be provided by the author of the importable units

Reviewer 2: ```suggestion
line, and should be provided by the author of the importable units
```
